aten/src/ATen/native/Resize.h

```diff
@@ -22,23 +22,24 @@ TORCH_API void resize_output(Tensor& output, IntArrayRef shape);
 // They are not in TH/THTensor.cpp because the at namespace is easier
 // to benchmark than TH; I can't get gbenchmark to call fns from THTensor.cpp
 
-static inline void maybe_resize_storage_cpu(TensorImpl* self, int64_t new_size) {
+static inline void maybe_resize_storage_cpu(TensorImpl* self, uint64_t new_size) {
```

Comment:
It makes sense that this function would take an unsigned integer, but this update doesn't seem to reflect the name of the PR.

This function's also exclusively called by resize_impl_cpu_, isn't the issue that it computes `storage_size` in int64_t and doesn't check for the size being representable in int64_t? Naively, I guess I would have expected an explicit check for new_size being less than zero or zero. 

Also, in the case where new_size == 0, what happens if the tensor doesn't have a storage pointer?

Reply:
Will update PR title (and also skip the test under asan, which correctly detects integer overflow).

And there is already a check for `new_size == 0`, which exists, according to https://github.com/pytorch/pytorch/commit/c46c6a4fe68a8ed6a8adb35837e87209cf020547  handle zero sized tensor slicing, i.e.:
```
python -c "import numpy as np;import torch;t=torch.from_numpy(np.empty((0,4)));t[:,1::2]*=1"
```

Reply:
I know there's a check for new_size == 0, but if the storage is uninitialized it just stays uninitialized in that case, right?

Reply:
Yes, as far as I understand, this is the intended behaviour.

Reply:
OK.